src/electron-main.ts

```diff
@@ -549,6 +560,51 @@ app.on("ready", async () => {
         global.mainWindow?.webContents.send("openDesktopCapturerSourcePicker");
         setDisplayMediaCallback(callback);
     });
+
+    session.defaultSession.webRequest.onBeforeRequest((req, callback) => {
+        // This handler emulates the element-web service worker, where URLs are rewritten late in the request
+        // for backwards compatibility. As authenticated media becomes more prevalent, this should be replaced
+        // by the app using authenticated URLs from the outset.
+        let url = req.url;
+        if (!url.includes("/_matrix/media/v3/download") && !url.includes("/_matrix/media/v3/thumbnail")) {
+            return callback({}); // not a URL we care about
+        }
+
+        // Check for feature support from the server. This requires asking the renderer process for supported
+        // versions.
+        ipcMain.once("serverSupportedVersions", (_, versionsResponse) => {
+            if (versionsResponse?.versions?.includes("v1.11")) {
+                url = url.replace(/\/media\/v3\/(.*)\//, "/client/v1/media/$1/");
+                return callback({ redirectURL: url });
+            } else {
+                return callback({}); // no support == no modification
+            }
+        });
```

Comment:
I think I'd prefer if the IPC was pushed to rather than queried twice on every matching media request

Reply:
How do you mean 'pushed'?

Reply:
Whenever the access token changes that change is shared via ipc with the main process, similar for the server versions which would be polled infrequently

Reply:
I was hoping to do that with the service worker stuff, but there didn't appear to be a reliable place for detecting that change and pushing it to places. It's further complicated by media requests happening *very* early in the app startup, where the service worker (or Electron) will be the first caller to cause `/versions` to be cached, for example.